StyleCop.Analyzers/StyleCop.Analyzers/OrderingRules/SA1214StaticReadonlyElementsMustAppearBeforeStaticNonReadonlyElements.cs

```diff
@@ -43,7 +46,61 @@ public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
         /// <inheritdoc/>
         public override void Initialize(AnalysisContext context)
         {
-            // TODO: Implement analysis
+            context.RegisterSyntaxNodeAction(this.AnalyzeClass, SyntaxKind.ClassDeclaration);
+            context.RegisterSyntaxNodeAction(this.AnalyzeStruct, SyntaxKind.StructDeclaration);
+        }
+
+        private void AnalyzeStruct(SyntaxNodeAnalysisContext context)
+        {
+            var structDeclaration = (StructDeclarationSyntax)context.Node;
+
+            AnalyzeType(context, structDeclaration);
+        }
+
+        private void AnalyzeClass(SyntaxNodeAnalysisContext context)
+        {
+            var classDeclaration = (ClassDeclarationSyntax) context.Node;
+
+            AnalyzeType(context, classDeclaration);
+        }
+
+        private static void AnalyzeType(SyntaxNodeAnalysisContext context, TypeDeclarationSyntax typeDeclaration)
+        {
+            var staticFields = typeDeclaration.Members
+                .OfType<BaseFieldDeclarationSyntax>()
+                .Where(f => f.Modifiers.Any(m => m.IsKind(SyntaxKind.StaticKeyword)))
+                .ToList();
+
+            var endLocationsOfStaticNonReadonlyFields =
+                staticFields.Where(f => f.Modifiers.Any(m => m.IsKind(SyntaxKind.ReadOnlyKeyword)) == false)
```

Comment:
To be honest I did it deliberately, because for me, when you look at !f.Modifiers.Any(m => m.IsKind(SyntaxKind.ReadOnlyKeyword)) you may omit ! at the beginning.